src/components/user/EmailChange.tsx

```diff
@@ -0,0 +1,99 @@
+import { SubmitHandler, useForm } from "react-hook-form";
+import { ChangeEmailFormInput } from "../../libs/types/user";
+import { emailRegex } from "../../libs/constants/regex";
+import { changeEmail } from "../../libs/user/user";
+import { useToast } from "../../layouts/AppProvider";
+
+const EmailChange = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setError,
+    reset,
+  } = useForm<ChangeEmailFormInput>();
+  const { showToast } = useToast();
+
+  const onSubmit: SubmitHandler<ChangeEmailFormInput> = async (data) => {
+    const response = await changeEmail(data);
+    if (response.status === true) {
+      showToast("Change email link sended.");
+      reset();
+    } else if ("errors" in response) {
+      if (response.errors) {
+        Object.keys(response.errors).forEach((field) => {
+          setError(field as keyof ChangeEmailFormInput, {
+            message: response.errors![field][0],
+          });
+        });
+      }
+    }
```

Comment:
Anh nghĩ viết như sau sẽ hợp lý hơn:

  if ("errors" in response && response.errors) {
      Object.keys(response.errors).forEach((field) => {
        setError(field as keyof ChangeEmailFormInput, {
          message: response.errors![field][0],
      });
    }
  }

  if (response.status === true) {
    showToast("Change email link sended.");
    reset();
  }